### ISSUE TITLE
Update controller for searching

### ISSUE THREAD

**Issue by adamnestor** [adamnestor/serviceTally#59]
- [ ] ServiceEventActivityController Update
- Add code to search for service events based on flags
- [ ] Add `@RequestParam` for each field I want to be able to filter the search with
- [ ] Add each of these params to the model
- A note I am considering is options for graduation year for instance to eliminate multiple ways of writing the same year for search purposes
- [ ] Add logic to check for search options
- [ ] Create methods: findAll and search
- [ ] Create methods in the repository
- [ ] Update service with methods

God bless this ticket

**Comment by adamnestor** [comment]
New plan that I believe works with updated desired functionality defined:

Desired functionality:

1. Retrieve and Display Students Names in a Dropdown
2. Filter by Student
3. Filter by Status

Backend Implementation

- [ ] Create a method to Retrieve Students
- [ ] Create Methods to Filter Service Events
- [ ] Update the Repository

Frontend Implementation

- [ ] Create Dropdown for Students
- [ ] Filter by Student or Status

Controller Implementation

- [ ] Create Endpoints for Students and Service Events

**Comment by adamnestor** [comment]
I have successfully added the service events to the appropriate dashboard. I tested a student changing their school and witnessed the service events leave one admin view and appear on the current updated view.

I believe it wise to split up this ticket. I will close this one having added the events to the dashboard. And will create a new ticket with the hopes of adding the search function as laid out above.
